app/javascript/gobierto_dashboards/modules/contracts_controller.js

```diff
@@ -117,24 +125,53 @@ export class ContractsController {
       }
     }
 
-    const result = {
+    // Contracts precalculations
+    _r = {
+      domain: [501, 1001, 5001, 10001, 15001],
+      range: [0, 1, 2, 3, 4, 5]
+    };
+    var rangeFormat = d3.scaleThreshold().domain(_r.domain).range(_r.range);
+
+    for(let i = 0; i < contractsData.length; i++){
+      const contract = contractsData[i],
+            final_amount = (contract.final_amount === '' || contract.final_amount === undefined) ? 0.0 : parseFloat(contract.final_amount);
+
+      contract.final_amount = final_amount;
+      contract.range = rangeFormat(+final_amount);
+    }
+
+    data = {
       contractsData: contractsData.sort(sortByField('end_date')),
       tendersData: tendersData.sort(sortByField('submission_date')),
     }
+  }
+
+  _renderSummary(){
+    ndx = crossfilter(data.contractsData);
 
-    return result;
+    this._renderTendersMetricsBox();
+    this._renderContractsMetricsBox();
+
+    this._renderByAmountsChart();
+    this._renderContractTypeChart();
+    this._renderProcessTypeChart();
   }
 
-  _renderSummary(remoteCsvData){
-    this._renderTendersMetricsBox(remoteCsvData.tendersData);
-    this._renderContractsMetricsBox(remoteCsvData.contractsData);
+  _refreshData(reducedContractsData){
+    reduced = {tendersData: data.tendersData, contractsData: reducedContractsData};
+
+    vueApp.contractsData = reducedContractsData;
```

Comment:
How is Vue able to access to this property?

Reply:
When the vue app is initialized, that's one of its data variables.

```javascript 
vueApp = new Vue({
  router,
  data: Object.assign(options, data),
}).$mount(entryPoint);
```

in `Object.assign(options, data)`, `data` is:

```javascript 
data = {
  contractsData: contractsData.sort(sortByField('end_date')),
  tendersData: tendersData.sort(sortByField('submission_date')),
}
```